version: '3.9'

services: 
    db:
        image: postgres
        volumes:
            - ./data/db:/var/lib/postgresql/data
        environment: 
            - POSTGRES_DB=postgres
            - POSTGRES_USER=postgres
            - POSTGRES_PASSWORD=postgres
        
    web:
        build: .
        command: python manage.py runserver 0.0.0.0:8000
        volumes:
            - .:/code
        ports:
            - "8000:8000"
        depends_on:
            - db
            - rabbit
            - celery-worker

    
    celery-worker:
        build: .
        command: celery -A practice worker -l INFO
        env_file:
            - .env
        volumes:
            - .:/code
            # - ./data/db:/var/lib/postgresql/data
        depends_on:
            - db
            - rabbit
    
    celery-beat:
        build: .
        command:  celery -A practice beat -l info --scheduler django_celery_beat.schedulers:DatabaseScheduler
        env_file:
            - .env
        volumes:
            - .:/code
            # - ./data/db:/var/lib/postgresql/data
        depends_on:
            - db
            - rabbit
            - celery-worker
    
        
    rabbit:
        image: "rabbitmq:3-management-alpine"
        env_file:
            - .env
        ports:
            - "5672:5672"
            - "15672:15672"
        restart: on-failure
    